src/strategies/uniswapx_strategy.rs

```diff
@@ -199,16 +203,45 @@ impl<M: Middleware + 'static> UniswapXUniswapFill<M> {
                 sig: Bytes::from_str(signature)?,
             });
         }
-        // abi encode as [tokens to approve, multicall data]
+        // slice off function selector
+        let cleaned_method_paramaters = &route.method_parameters.calldata[10..];
+        info!("cleaned_method_paramaters: {}", cleaned_method_paramaters);
+
+        let mut tokens_to_approve_to_swap_router02 = Vec::new();
+        let mut tokens_to_approve_to_reactor = Vec::new();
+        // check token approvals
+        let token_in_contract = ERC20::new(H160::from_str(&request.token_in)?, self.client.clone());
+        let token_out_contract = ERC20::new(H160::from_str(&request.token_out)?, self.client.clone());
+        let allowance_in = token_in_contract
+            .allowance(
+                H160::from_str(EXECUTOR_ADDRESS)?,
+                H160::from_str(SWAPROUTER_02_ADDRESS)?,
+            )
+            .await?;
```

Comment:
this is a little hard to read so I might prefer a helper funciton so its more like
```
let router_approvals = if needs_approval(token_in) {
   vec![token_in]
} else {
  vec![]
}
```